src/antsibull/cli/antsibull_build.py

```diff
@@ -247,6 +258,17 @@ def parse_args(program_name: str, args: list[str]) -> argparse.Namespace:
         " as well as changelog and porting guide if applicable."
         "  Defaults to --data-dir",
     )
+    constraints_parser = argparse.ArgumentParser(add_help=False)
+    constraints_parser.add_argument(
+        "--constraints-file",
```

Comment:
I'm pondering whether we should always honor a `ansible-MAJOR_VERSION.constraints` file if it exists in the data_dir instead of requiring this option. Then, we can remove the conditionals in the release playbook and the build-dumb-pypi workflow that check for its existence.

Reply:
I think this would be better.